packages/xo-server/src/api/vif.js

```diff
@@ -85,6 +85,15 @@ export async function set({
     push.apply(newIpAddresses, allowedIpv6Addresses || vif.allowedIpv6Addresses)
   }
 
+  lockingMode =
```

Comment:
```js
if(lockingMode !== undefined) {
  await this.hasPermissions(this.user.id, [
    [vif.$VM, 'operate'],
    [network?.id ?? vif.$network, 'operate'],
  ])
}
```

Reply:
But the `hasPermissions` will return `false` if there is no permission.

Reply:
IMHO, we need to inform the user if he doesn't have permission to do an operation, so, we can throw in this case. To do that we can use `checkPermissions` instead.